packages/kit/src/runtime/client/renderer.js

```diff
@@ -615,6 +615,9 @@ export class Renderer {
 		if (props) {
 			// shadow endpoint props means we need to mark this URL as a dependency of itself
 			node.uses.dependencies.add(url.href);
+			for (const key in params) {
```

Comment:
This 3 lines may be redundant because we are checking `has_shadow` instead of uses.dependencies, but maybe for being safe, in case we need this info for something else.

Reply:
By 3 lines I mean 618 and the 2 below it

Reply:
@Rich-Harris suggested we might not want to automatically add all params because while it makes sense in isolation, it may be too eager if we add a layout equivalent of shadow endpoints

Reply:
@benmccann Yeah, I understand, but what do you suggest, how do we deal with that now, maybe add some check if it's the last branch, or something like that, 
but when/if you add layout shadows, how would you track the params that are used then?